packages/react-core/src/components/Drawer/Drawer.tsx

```diff
@@ -31,23 +30,21 @@ export const DrawerContext = React.createContext<Partial<DrawerContextProps>>({
   isStatic: false
 });
 
+const timeout = Number.parseInt(transitionDuration.value.match(/\d+/)[0]);
+
 export const Drawer: React.SFC<DrawerProps> = ({
   className = '',
   children,
   isExpanded = false,
   isInline = false,
   isStatic = false,
   position = 'right',
-  onMount = () => {},
-  onUnmount = () => {},
+  onExpand = () => {},
   ...props
 }: DrawerProps) => {
-  React.useEffect(() => {
-    onMount();
-    return () => {
-      onUnmount();
-    };
-  }, [isExpanded, onMount, onUnmount]);
+  if (isExpanded) {
+    setTimeout(onExpand, timeout);
```

Comment:
Don't like seeing setTimeout because the duration may not always be enough. 

Would it be possible to use something like ResizeObserver to call onExpand upon a resize event? Assuming there is an event to use...

Reply:
If the animation takes longer than `250ms` this will cause some jerkiness on the left side equivalent to the difference between the CSS `animation-delay` and `setTimeout`, but the code will still work.

AFAIK no resize event is being called. This is the core CSS:
```css
@keyframes pf-remove-tab-focus {
  to {
    visibility: hidden; } }
.pf-c-drawer__panel[hidden] {
  animation-name: pf-remove-tab-focus;
  animation-delay: var(--pf-c-drawer__panel--TransitionDuration);
  animation-fill-mode: forwards; }
```

I'm happy to use observers if there's something to observe.

Reply:
Yesterday @jschuler found the [onTransitionEnd](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/transitionend_event) event and I've switched to using that. Please rereview @dlabrecq !